### ISSUE TITLE
Incorrect .pat reading

### ISSUE THREAD

**Issue by hertz4** [milkytracker/MilkyTracker#51]
Most pat files don't read correctly.

![2016-09-12-221215_938x109_scrot](https://cloud.githubusercontent.com/assets/11461551/18460316/12ac2f34-7936-11e6-9a78-8ae35ddda29a.png)

Here's an example. Playing a scale here results in some random-sounding pitches, though some of it is definitely correct there are also errors. Here's the offending .pat file:
[https://drive.google.com/file/d/0B-NDUlDEAog8WjZkOVotQ1ZIeUU/view?usp=sharing](url)

Most freepats I've tried act the same way. Often, sample names become corrupted as well. I've even had milkytracker lock up completely when reading a .pat, though I can't seem to reproduce that consistently.

**Comment by milkypailes** [comment]
Did somebody download the file, it's no longer available.

**Comment by hertz4** [comment]
umm, what the heck? The file is still there, but the link gives a github 404. If you copy-paste the URL, you can still get it.

**Comment by hertz4** [comment]
[pat.zip](https://github.com/milkytracker/MilkyTracker/files/855900/pat.zip)

**Comment by hisojarvi** [comment]
I made some changes to how GUS frequency ranges are mapped to instrument notes. I was able to fix the gaps where some notes between 1 and 2 were mapped to sample 0 on the keyboard. Also most of the relative notes are now correct. 

One outstanding issue is that the relative note of sample smp07m in Accordian.pat is still incorrect, but I am not sure whether it's an issue in the code or in the instrument data. The sample has root frequency 1479978, which is one less than 1479979, which is one of the frequencies in the conversion table. 

I also tested the changes with many other patches, and they seem to load more correctly than before:
https://www.doomworld.com/idgames/music/dgguspat
https://packages.debian.org/sid/sound/freepats
